zaza/openstack/charm_tests/saml_mellon/tests.py

```diff
@@ -156,3 +158,177 @@ def _do_redirect_check(url, region, idp_expect, horizon_expect):
         # We may need to try/except to allow horizon to build its pages
         _do_redirect_check(url, region, idp_expect, horizon_expect)
         logging.info("SUCCESS")
+
+
+class BaseCharmKeystoneSAMLMellonTest(BaseKeystoneTest):
+    """Charm Keystone SAML Mellon tests."""
+
+    @classmethod
+    def setUpClass(cls,
+                   application_name="keystone-saml-mellon",
+                   test_saml_idp_app_name="test-saml-idp",
+                   horizon_idp_option_name="myidp_mapped",
+                   horizon_idp_display_name="myidp via mapped"):
+        """Run class setup for running Keystone SAML Mellon charm tests."""
+        super(BaseCharmKeystoneSAMLMellonTest, cls).setUpClass()
+        cls.application_name = application_name
+        cls.test_saml_idp_app_name = test_saml_idp_app_name
+        cls.horizon_idp_option_name = horizon_idp_option_name
+        cls.horizon_idp_display_name = horizon_idp_display_name
+        cls.action = "get-sp-metadata"
+        cls.current_release = openstack_utils.get_os_release()
+        cls.FOCAL_USSURI = openstack_utils.get_os_release("focal_ussuri")
+
+    def test_run_get_sp_metadata_action(self):
+        """Validate the get-sp-metadata action."""
+        unit = zaza.model.get_units(self.application_name)[0]
+        if self.vip:
+            ip = self.vip
+        else:
+            ip = unit.public_address
```

Comment:
I tend to prefer (but don't change it if you prefer the current format):

```python
ip = self.vip if self.vip else unit.public_address
```

Reply:
I like this as well. I'll change it.